src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -1065,5 +1070,35 @@ private void LogErrorFromException(Exception exception)
                     break;
             }
         }
+
+        /// <summary>
+        /// Logs the list of files to embed in the MSBuild binary log.
+        /// </summary>
+        /// <param name="dependencyGraphSpec"></param>
+        private void LogFilesToEmbedInBinlog(DependencyGraphSpec dependencyGraphSpec)
+        {
+            // If the MSBuildBinaryLoggerEnabled environment variable is not set, don't log the paths to the files.
+            if (!string.Equals(Environment.GetEnvironmentVariable("MSBUILDBINARYLOGGERENABLED"), bool.TrueString, StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            IReadOnlyList<PackageSpec> projects = dependencyGraphSpec.Projects;
+
+            foreach (PackageSpec project in projects)
+            {
+                if (project.RestoreMetadata.ProjectStyle == ProjectStyle.PackageReference)
+                {
+                    LoggingQueue.Enqueue(new ConsoleOutLogEmbedInBinlog(Path.Combine(project.RestoreMetadata.OutputPath, LockFileFormat.AssetsFileName)));
+                    LoggingQueue.Enqueue(new ConsoleOutLogEmbedInBinlog(Path.Combine(project.RestoreMetadata.OutputPath, DependencyGraphSpec.GetDGSpecFileName(Path.GetFileName(project.RestoreMetadata.ProjectPath)))));
+                    LoggingQueue.Enqueue(new ConsoleOutLogEmbedInBinlog(BuildAssetsUtils.GetMSBuildFilePathForPackageReferenceStyleProject(project, BuildAssetsUtils.PropsExtension)));
+                    LoggingQueue.Enqueue(new ConsoleOutLogEmbedInBinlog(BuildAssetsUtils.GetMSBuildFilePathForPackageReferenceStyleProject(project, BuildAssetsUtils.TargetsExtension)));
+                }
+                else if (project.RestoreMetadata.ProjectStyle == ProjectStyle.PackagesConfig)
+                {
+                    LoggingQueue.Enqueue(new ConsoleOutLogEmbedInBinlog(Path.Combine(Path.GetDirectoryName(project.RestoreMetadata.ProjectPath), NuGetConstants.PackageReferenceFile)));
```

Comment:
`NuGetConstants.PackageReferenceFile` is a constant with value `packages.config`. However, projects can also use `packages.<project name>.config`: https://github.com/NuGet/NuGet.Client/blob/a897a1478161a2b5cbf730879977cab3a0b1cef9/src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs#L428-L440